//cmd/gazelle:go_default_test is failing on master

```
    TestMatchProtoLibrary: integration_test.go:3010: proto/BUILD.bazel: got:
        load("@rules_proto//proto:defs.bzl", "proto_library")
        # gazelle:prefix example.com/foo
        
        some_rule(
            name = "gen_proto",
            out = "foo.proto",
        )
        
        proto_library(
            name = "existing_proto",
            srcs = ["foo.proto"],
        )
        
        want:
        
        load("@io_bazel_rules_go//go:def.bzl", "go_library")
        load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
        load("@rules_proto//proto:defs.bzl", "proto_library")
        # gazelle:prefix example.com/foo
        
        some_rule(
            name = "gen_proto",
            out = "foo.proto",
        )
        
        proto_library(
            name = "existing_proto",
            srcs = ["foo.proto"],
            visibility = ["//visibility:public"],
        )
        
        go_proto_library(
            name = "proto_go_proto",
            importpath = "example.com/foo",
            proto = ":existing_proto",
            visibility = ["//visibility:public"],
        )
        
        go_library(
            name = "go_default_library",
            embed = [":proto_go_proto"],
            importpath = "example.com/foo",
            visibility = ["//visibility:public"],
        )
--- FAIL: TestMatchProtoLibrary (0.00s)
```

Started on #730. I'll roll it back for now.

@linzhp, would you mind updating the test and send another PR? I think it just wasn't updated after #733.

--- a/cmd/gazelle/integration_test.go
+++ b/cmd/gazelle/integration_test.go
@@ -2987,16 +2987,15 @@ func TestMatchProtoLibrary(t *testing.T) {
 load("@rules_proto//proto:defs.bzl", "proto_library")
 # gazelle:prefix example.com/foo
 
-some_rule(
-	name = "gen_proto",
-	out = "foo.proto",
-)
-
 proto_library(
 	name = "existing_proto",
 	srcs = ["foo.proto"],
 )
 `,
+		},
+		{
+			Path:    "proto/foo.proto",
+			Content: `syntax = "proto3";`,
 		},
 	}
 	dir, cleanup := testtools.CreateFiles(t, files)
@@ -3016,11 +3015,6 @@ load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
 load("@rules_proto//proto:defs.bzl", "proto_library")
 # gazelle:prefix example.com/foo
 
-some_rule(
-    name = "gen_proto",
-    out = "foo.proto",
-)
-
 proto_library(
     name = "existing_proto",
     srcs = ["foo.proto"],
@@ -3028,7 +3022,7 @@ proto_library(
 )
 
 go_proto_library(
-    name = "proto_go_proto",
+    name = "foo_go_proto",
     importpath = "example.com/foo",
     proto = ":existing_proto",
     visibility = ["//visibility:public"],
@@ -3036,11 +3030,10 @@ go_proto_library(
 
 go_library(
     name = "go_default_library",
-    embed = [":proto_go_proto"],
+    embed = [":foo_go_proto"],
     importpath = "example.com/foo",
     visibility = ["//visibility:public"],
-)
-`,
+)`,
 		},
 	})
 }

--- a/cmd/gazelle/integration_test.go
+++ b/cmd/gazelle/integration_test.go
@@ -2975,3 +2975,72 @@ go_repository(
 		},
 	})
 }
+
+func TestMatchProtoLibrary(t *testing.T) {
+	files := []testtools.FileSpec{
+		{
+			Path: "WORKSPACE",
+		},
+		{
+			Path: "proto/BUILD.bazel",
+			Content: `
+load("@rules_proto//proto:defs.bzl", "proto_library")
+# gazelle:prefix example.com/foo
+
+some_rule(
+	name = "gen_proto",
+	out = "foo.proto",
+)
+
+proto_library(
+	name = "existing_proto",
+	srcs = ["foo.proto"],
+)
+`,
+		},
+	}
+	dir, cleanup := testtools.CreateFiles(t, files)
+	defer cleanup()
+
+	args := []string{"update"}
+	if err := runGazelle(dir, args); err != nil {
+		t.Fatal(err)
+	}
+
+	testtools.CheckFiles(t, dir, []testtools.FileSpec{
+		{
+			Path: "proto/BUILD.bazel",
+			Content: `
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
+load("@rules_proto//proto:defs.bzl", "proto_library")
+# gazelle:prefix example.com/foo
+
+some_rule(
+    name = "gen_proto",
+    out = "foo.proto",
+)
+
+proto_library(
+    name = "existing_proto",
+    srcs = ["foo.proto"],
+    visibility = ["//visibility:public"],
+)
+
+go_proto_library(
+    name = "proto_go_proto",
+    importpath = "example.com/foo",
+    proto = ":existing_proto",
+    visibility = ["//visibility:public"],
+)
+
+go_library(
+    name = "go_default_library",
+    embed = [":proto_go_proto"],
+    importpath = "example.com/foo",
+    visibility = ["//visibility:public"],
+)
+`,
+		},
+	})
+}

--- a/language/proto/kinds.go
+++ b/language/proto/kinds.go
@@ -19,6 +19,7 @@ import "github.com/bazelbuild/bazel-gazelle/rule"
 
 var protoKinds = map[string]rule.KindInfo{
 	"proto_library": {
+		MatchAttrs:    []string{"srcs"},
 		NonEmptyAttrs: map[string]bool{"srcs": true},
 		MergeableAttrs: map[string]bool{
 			"srcs": true,

--- a/merger/merger.go
+++ b/merger/merger.go
@@ -38,6 +38,7 @@ package merger
 
 import (
 	"fmt"
+	"sort"
 	"strings"
 
 	"github.com/bazelbuild/bazel-gazelle/rule"
@@ -222,19 +223,15 @@ func Match(rules []*rule.Rule, x *rule.Rule, info rule.KindInfo) (*rule.Rule, er
 
 	for _, key := range info.MatchAttrs {
 		var attrMatches []*rule.Rule
-		xvalue := x.AttrString(key)
-		if xvalue == "" {
-			continue
-		}
 		for _, y := range kindMatches {
-			if xvalue == y.AttrString(key) {
+			if attrMatch(x, y, key) {
 				attrMatches = append(attrMatches, y)
 			}
 		}
 		if len(attrMatches) == 1 {
 			return attrMatches[0], nil
 		} else if len(attrMatches) > 1 {
-			return nil, fmt.Errorf("could not merge %s(%s): multiple rules have the same attribute %s = %q", xkind, xname, key, xvalue)
+			return nil, fmt.Errorf("could not merge %s(%s): multiple rules have the same attribute %s", xkind, xname, key)
 		}
 	}
 
@@ -247,4 +244,24 @@ func Match(rules []*rule.Rule, x *rule.Rule, info rule.KindInfo) (*rule.Rule, er
 	}
 
 	return nil, nil
+}
+
+func attrMatch(x, y *rule.Rule, key string) bool {
+	xValue := x.AttrString(key)
+	if xValue != "" && xValue == y.AttrString(key) {
+		return true
+	}
+	xValues := x.AttrStrings(key)
+	yValues := y.AttrStrings(key)
+	if xValues == nil || yValues == nil || len(xValues) != len(yValues) {
+		return false
+	}
+	sort.Strings(xValues)
+	sort.Strings(yValues)
+	for i, v := range xValues {
+		if v != yValues[i] {
+			return false
+		}
+	}
+	return true
 }

--- a/merger/merger_test.go
+++ b/merger/merger_test.go
@@ -969,6 +969,15 @@ go_binary(name = "y")
 go_binary(name = "z")
 `,
 			wantError: true,
+		}, {
+			desc:      "srcs match",
+			gen:       `proto_library(name = "proto1", srcs = ["foo.proto", "bar.proto"])`,
+			old:       `proto_library(name = "proto2", srcs = ["bar.proto", "foo.proto"])`,
+			wantIndex: 0,
+		}, {
+			desc: "importpath match",
+			gen:  `go_proto_library(name = "go_proto1", importpath="example.com/foo")`,
+			old:  `go_proto_library(name = "go_proto2", importpath="example.com/foo")`,
 		},
 	} {
 		t.Run(tc.desc, func(t *testing.T) {

